lib/thrift_serializer/encoder.ex

```diff
@@ -0,0 +1,26 @@
+defmodule ThriftSerializer.Encoder do
+
+  def encode(struct, model, thrift_module) do
+    struct_definition = {:struct, {thrift_module, model}}
+
+    try do
+      with {:ok, tf} <- :thrift_memory_buffer.new_transport_factory(),
+           {:ok, pf} <- :thrift_binary_protocol.new_protocol_factory(tf, []),
+           {:ok, binary_protocol} <- pf.()
```

Comment:
Add a private function that extracts this method. The following should probably work:

```
defp binary_protocol do
  {:ok, tf} <- :thrift_memory_buffer.new_transport_factory(),
  {:ok, pf} <- :thrift_binary_protocol.new_protocol_factory(tf, []),
  {:ok, binary_protocol} <- pf.()

  binary_protocol
end
```